### PR TITLE
Copy Snapshot operation arguments in a distributed task

### LINKED ISSUE #10390: Wrong file names created in distributed Snapshot
### Describe the bug

Running a `Snapshot` in distributed mode creates as many partial snapshots as distributed tasks. Each task will append its task id to the original file name, so that different tasks do not overwrite each other's data. The input file name is modified in-place in each task, but this leads to a situation where a task can receive a modified file name and modify it further, thus leading to wrong file names after the Snapshot. The following reproducer
```python
import ROOT
RDataFrame = ROOT.RDF.Experimental.Distributed.Dask.RDataFrame

from dask.distributed import Client, LocalCluster

import os

def test(client):

    df = RDataFrame(10, daskclient=client).Define("a","1.")

    snap_df = df.Snapshot("dummy_distributed", "dummy_distributed.root")
    print(snap_df._headnode.inputfiles)

    tmp_files = [f"dummy_distributed_{i}.root" for i in range(2)]

if __name__ == "__main__":
    client = Client(LocalCluster(n_workers=2, threads_per_worker=1, processes=True))
    for i in range(2):
        test(client)
```
leads to
```
$: python repro.py
['dummy_distributed_0.root', 'dummy_distributed_1.root']
['dummy_distributed_1_0.root', 'dummy_distributed_0_1.root']
```

Interestingly, this happens only with Dask

### Expected behavior
Each task can properly process the Snapshot in isolation.

### Setup
ROOT >= 6.26

### PR DESCRIPTION
**Pull request by vepadulano**
The Snapshot operation file name is modified in-place to append the
range id of a certain task. This can lead to a task receiving the
input operation from a previous task with an already modified file
name. Thus, the current task would create a wrong file name with more
than one range id. Solve this by creating a deep copy of the Snapshot
operation arguments in each task, so that the filename is correctly
changed in isolation.

This PR fixes #10390

### CODE REVIEW

**Comment by vepadulano** [comment]
@phsft-bot build also on mac12/default

**Comment by vepadulano** [comment]
@phsft-bot build

**Comment by vepadulano** [comment]
@phsft-bot build

File: bindings/experimental/distrdf/python/DistRDF/ComputationGraphGenerator.py [hunk 1]
```diff
@@ -120,16 +144,16 @@ def _call_rdf_operation(distrdf_node: Node, previous_rdf_node: Any, range_id: in
     future_results = []
 
     rdf_operation = getattr(previous_rdf_node, distrdf_node.operation.name)
-    _make_op_lazy_if_needed(distrdf_node.operation, range_id)
-    pyroot_node = rdf_operation(*distrdf_node.operation.args, **distrdf_node.operation.kwargs)
+    in_task_op = _create_lazy_op_if_needed(distrdf_node.operation, range_id)
```
Comment (etejedor) [inline bindings/experimental/distrdf/python/DistRDF/ComputationGraphGenerator.py:147]: So two tasks can invoke `_call_rdf_operation` on the same `distrdf_node` object? How does this happen, I thought every task generates its own graph?

Comment (vepadulano) [reply bindings/experimental/distrdf/python/DistRDF/ComputationGraphGenerator.py:147]: Every task generates its own RDataFrame C++ graph, but the DistRDF Python graph is a single object that gets serialized/deserialized. On a single machine, a single Python process can receive two(or more) tasks. When the first task starts, it deserializes the distrdf graph, modifies *in-place* the `operation` object, then it creates the RDF C++ calls, sends everything to the mapper function that executes them. When the second task starts, it gets the same distrdf graph objects,  but at this point their `operation` attributes were modified by the previous task, thus leading to the errors described in the PR

Comment (etejedor) [reply bindings/experimental/distrdf/python/DistRDF/ComputationGraphGenerator.py:147]: I see, thanks!

**Review by etejedor — APPROVED** [review]

**Pull request merged**